lib/pharos/phase.rb

```diff
@@ -79,6 +79,7 @@ def host_configurer
     # @return [K8s::Client]
     def kube_client
       fail "Phase #{self.class.name} does not have kube @master" unless @master
+      fail "Phase #{self.class.name} does not have kubeconfig cluster_context" unless cluster_context['kubeconfig']
 
       @kube_client ||= Pharos::Kube.client(@master.api_address, cluster_context['kubeconfig'])
```

Comment:
Note that the value for `@master.api_address` being passed here is completely irrelevant now, it could be anything at all.

Reply:
The refactoring here could be taken a step further to replace the `master` param in the phases with a `kube_client`, but I'm not sure I want to do that in the same PR.

For the `Pharos::Addon`, it happened here because the class would have needed a new `cluster_context` or `kube_config` param as well, and it made more sense to just go all the way and pass the `kube_client` instead.